src/shared/components/palette/case-flag/components/select-flag-type/select-flag-type.component.ts

```diff
@@ -21,37 +25,70 @@ export class SelectFlagTypeComponent implements OnInit {
   public flagCommentsOptionalEmitter: EventEmitter<any> = new EventEmitter();
 
   public flagTypes: FlagType[];
-  public flagTypeSelected: string;
+  public selectedFlagType: FlagType;
   public errorMessages: ErrorMessage[];
   public flagTypeNotSelectedErrorMessage = '';
   public flagTypeErrorMessage = '';
+  public flagRefdata$: Subscription;
+  public otherFlagTypeSelected = false;
+  public refdataError = false;
 
+  public readonly flagTypeControlName = 'flagType';
+  public readonly descriptionControlName = 'otherFlagTypeDescription';
   private readonly maxCharactersForOtherFlagType = 80;
+  // Code for "Other" flag type as defined in Reference Data
+  private readonly otherFlagTypeCode = 'OT0001';
 
   public get caseFlagWizardStepTitle(): typeof CaseFlagWizardStepTitle {
     return CaseFlagWizardStepTitle
   }
 
+  constructor(private readonly caseFlagRefdataService: CaseFlagRefdataService) { }
+
   public ngOnInit(): void {
-    this.flagTypes = this.getFlagTypes();
-    this.formGroup.addControl('flagType', new FormControl(''));
-    this.formGroup.addControl('other', new FormControl(''));
-    this.formGroup.addControl('otherFlagTypeDescription', new FormControl(''));
+    this.flagTypes = [];
+    this.formGroup.addControl(this.flagTypeControlName, new FormControl(''));
+    this.formGroup.addControl(this.descriptionControlName, new FormControl(''));
+
+    // HMCTS service code should come from Refdata API lookup via getHmctsServiceCode() but this is not working in AAT;
+    // use test service code 'AAA1' for now
+    this.flagRefdata$ = this.caseFlagRefdataService.getCaseFlagsRefdata('AAA1', RefdataCaseFlagType.PARTY).subscribe({
```

Comment:
What does this 'AAA1' hard code mean?

Reply:
@rayliangatos - 'AAA1' is a test service code to be used for looking up Case Flags refdata. It will be replaced in a subsequent PR.

Reply:
@Daniel-Lam okay, as I'm seeting this  'AAA1' is added to the actual logical ts file not the spec.ts test file, which confused me and I think it should be avoided, And it would be better to passed from somewhere correctly.